crates/ruff_db/src/diagnostic/mod.rs

```diff
@@ -868,30 +880,51 @@ pub enum DiagnosticTag {
 /// be in kebab case, e.g. `no-foo` (all lower case).
 ///
 /// Rules use kebab case, e.g. `no-foo`.
-#[derive(Debug, Copy, Clone, PartialOrd, Ord, PartialEq, Eq, Hash, get_size2::GetSize)]
-pub struct LintName(&'static str);
```

Comment:
It's not the end of the world but it seems unfortunate if we have to change such fundamental types for an experimental feature. 

The alternative here would be to retrieve the lint registry from the database (or a thread local) during deserialization and to use it to resolve the lint metadata. 

https://github.com/astral-sh/ruff/blob/99d0ac60b417ee623bb9257a6502a7c21c444b28/crates/ty_python_semantic/src/db.rs#L14